extensions/ql-vscode/src/quickEvalCodeLensProvider.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  CodeLensProvider,
+  TextDocument,
+  CodeLens, 
+  Command,
+  Range
+} from 'vscode';
+  
+class QuickEvalCodeLensProvider implements CodeLensProvider {
+  // Each provider requires a provideCodeLenses function which will give the various documents
+  // the code lenses
+  async provideCodeLenses(document: TextDocument): Promise<CodeLens[]> {
+
+    const codeLenses: CodeLens[] = [];
+
+    for (let index = 0; index < document.lineCount; index++) {
+      const textLine = document.lineAt(index);
+      // Match a predicate signature, including predicate name, parameter list, and opening brace.
+      const regex = new RegExp(/(\w+)\s*\(\s*.*(?:,\s*)*\)\s*\{/);
+      const matches = textLine.text.match(regex);
+
+      if (matches) {
+        const range: Range = new Range(
+          textLine.range.start.line, matches.index!,
+          textLine.range.end.line, matches.index! + 1
+        );
+        console.log(range);
+
+        const command: Command = {
+          command: 'codeQL.codeLensQuickEval',
+          title: `Quick Evaluation: ${matches![1]}`,
```

Comment:
Type assertion no longer necessary.

```suggestion
          title: `Quick Evaluation: ${matches[1]}`,
```

Reply:
Change has been made.